### PR TITLE
GLTFLoader: Fix loadAnimation for Morph + MultiMaterial

### PR DESCRIPTION
**Pull request by takahirox**
I should've updated  `loadAnimation` for MultiMaterial.

### CODE REVIEW

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -2841,20 +2841,34 @@ THREE.GLTFLoader = ( function () {
...
 							// PATH_PROPERTIES.weights(morphTargetInfluences) should be
-							// the property of a mesh object under node.
-							// So finding targets here.
+							// the property of a mesh object under group.
 
-							node.traverse( function ( object ) {
+							var children = node.isGroup ? node.children : [ node ];
 
-								if ( object.isMesh === true && object.material.morphTargets === true ) {
+							for ( var j = 0, jl = children.length; j < jl; j ++ ) {
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: Maybe just keep the `node.traverse` here? I find that much easier to read.

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -2841,20 +2841,34 @@ THREE.GLTFLoader = ( function () {
...
+									for ( var k = 0, kl = materials.length; k < kl; k ++ ) {
+
+										if ( materials[ k ].morphTargets === true ) {
+
+											targetNames.push( children[ j ].name ? children[ j ].name : children[ j ].uuid );
+											break;
+
+										}
+
+									}
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: Alternatively:

```js
if ( materials.some( function ( m ) { return m.morphTargets === true; } ) ) {

  targetNames.push( children[ j ].name ? children[ j ].name : children[ j ].uuid );

}
```

Comment (takahirox) [reply examples/js/loaders/GLTFLoader.js]: Wondering if I can just check if `.morphTargetInfluences.length` > 0. If I can, it'd be simpler.

**Comment by takahirox** [comment]
Updated. The change has become much simpler.

**Review by donmccurdy — APPROVED** [review]

**Comment by donmccurdy** [comment]
Nice! Yes that's much nicer. 😁

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**